plugins/outputs/cloudwatchlogs/pusher.go

```diff
@@ -55,29 +68,56 @@ type pusher struct {
 	stop                <-chan struct{}
 	lastSentTime        time.Time
 
+	backlogQueue        persistentqueue.PersistentQueue
+	ticker              *time.Ticker
+	dequeueEvents       *retryStruct
+	flagPersistentQueue bool
+
 	initNonBlockingChOnce sync.Once
 	startNonBlockCh       chan struct{}
 	wg                    *sync.WaitGroup
 	agentInfo             agentinfo.AgentInfo
 }
 
-func NewPusher(target Target, service CloudWatchLogsService, flushTimeout time.Duration, retryDuration time.Duration, logger telegraf.Logger, stop <-chan struct{}, wg *sync.WaitGroup, agentInfo agentinfo.AgentInfo) *pusher {
+func NewPusher(target Target, service CloudWatchLogsService, flushTimeout time.Duration, retryDuration time.Duration, logger telegraf.Logger, stop <-chan struct{}, wg *sync.WaitGroup, agentInfo agentinfo.AgentInfo, flagPersistentQueue bool) *pusher {
+	marshaler := func(i interface{}) ([]byte, error) { return json.Marshal(i) }
```

Comment:
I don't think we need to wrap this in another function. I think that you could just refer directly to `json.Marshal` for the queue's marshaller function

Reply:
yeah, you are right, panda, I don't need to wrap this in another function. I will change it right now.

Reply:
but if i refer directly to`json.Marshal` without wrap in `marshaler` , the `ummarshaler` function left there will  seems like not paired , right ? how do you think, panda, do I still need to change it? I am not sure now.

Reply:
Not sure what you mean? I don't think they need to be mirrored. If the marshaller function can be simplified to just call `json.Marshal` directly, we should do that. The reason the other function needs to be wrapped is because the function doesn't exactly match what we want.